pkg/asset/internal/templates.go

```diff
@@ -194,20 +200,29 @@ spec:
         - --root-ca-file=/etc/kubernetes/secrets/ca.crt
         - --service-account-private-key-file=/etc/kubernetes/secrets/service-account.key
         - --leader-elect=true
+        {{ with .CloudProvider -}}
+        - --cloud-provider={{ . }}
+        {{ end -}}
```

Comment:
Should also add `- --configure-cloud-routes=false` which is safe to always be set (we assume flannel will handle managing routes).